ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -634,9 +674,14 @@ public HttpContent readChunk(ChannelHandlerContext ctx) {
       if (chunk != null) {
         chunk.onDequeue();
         ByteBuf buf = Unpooled.wrappedBuffer(chunk.buffer);
-        chunk.writeCompleteThreshold = totalBytesWritten.addAndGet(chunk.bytesToBeWritten);
         chunksAwaitingCallback.add(chunk);
-        content = new DefaultHttpContent(buf);
+        if (contentLength != -1 && chunk.writeCompleteThreshold >= contentLength) {
```

Comment:
what do you think about making `isLastChunk` a property of the Chunk itself (which can figure it out in its constructor)?